packages/xo-server/src/api/vif.js

```diff
@@ -92,13 +106,24 @@ export async function set({
       mac,
       currently_attached: attached,
       ipv4_allowed: newIpAddresses,
+      locking_mode: lockingMode,
       qos_algorithm_type: rateLimit != null ? 'ratelimit' : undefined,
       qos_algorithm_params:
         rateLimit != null ? { kbps: String(rateLimit) } : undefined,
     })
 
     await this.allocIpAddresses(newVif.$id, newIpAddresses)
 
+    if (
+      lockingMode === 'network_default' &&
+      network !== undefined &&
+      networkDefaultIsLocked !== undefined
+    ) {
+      await network.set_default_locking_mode(
```

Comment:
IMO, the network default locking mode should only be editable by editing the network, not the VIF. (and only from the network view in the UI)